File: money-splitter-core/src/main/java/org/example/moneysplitter/core/entities/Spending.java
package org.example.moneysplitter.core.entities;

import lombok.Builder;
import lombok.Value;

import java.math.BigDecimal;
import java.util.Map;

@Value
@Builder
public class Spending {
    Participant payer;
    String product;
    Map<Participant, BigDecimal> proportions;
}

File: frontend/src/models/Spending.ts
import type {Participant} from "@/models/Participant.ts";
import type {Proportion} from "@/models/Proportion.ts";
import type {SplitType} from "@/models/SplitType.ts";

export class Spending {

  id: string;

  payer: Participant;

  name: string;

  amount: number;

  splitType: SplitType;

  proportions: Proportion[];
}